core/iwasm/aot/aot_loader.c

```diff
@@ -2764,6 +2764,57 @@ load_relocation_section(const uint8 *buf, const uint8 *buf_end,
     return ret;
 }
 
+#if WASM_ENABLE_LINUX_PERF != 0
+static uintptr_t
+get_func_addr(const AOTModule *module, uint32 func_idx_adj)
+{
+    return (uintptr_t)module->func_ptrs[func_idx_adj];
+}
+
+static uint32
+get_func_size(const AOTModule *module, uint32 func_idx_adj)
+{
+    uint32 func_sz;
+    if (func_idx_adj + 1 < module->func_count) {
+        func_sz = module->func_ptrs[func_idx_adj + 1]
+                  - module->func_ptrs[func_idx_adj];
+    }
+    else {
+        func_sz = module->code_size
+                  - (module->func_ptrs[func_idx_adj] - module->code);
+    }
```

Comment:
An issue is that the function pointers may be not ordered, e.g. when AOT static PGO is enabled. Had better traverse all the pointers and get a pointer which is larger than current pointer and is the smallest.